### PR TITLE
packaging: fix image build script

### PR DESCRIPTION
**Pull request by egernst**
There were a couple of issues with the build-scripts discovered while
doing release:
 - Relative paths are error prone. Fix error.
 - short_commit_length is used to truncate sha for commits when
 appending agent version to resulting files. Before this was
 in pkglib.sh, which is otherwise an unused file from when we
 supported OBS. Add this define to lib.sh, which is sourced by
 the applicable packaging scripts.

There's plenty of room for improvement, but these fixes make the
existing scripts functional again.

Fixes: #871

### CODE REVIEW

File: tools/packaging/scripts/lib.sh [hunk 1]
```diff
@@ -11,6 +11,8 @@ export tests_repo_dir="$GOPATH/src/$tests_repo"
 
 this_script_dir="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
 
+short_commit_length=10
```
Comment (jodh-intel) [inline tools/packaging/scripts/lib.sh:14]: Could you put this on a separate commit and explain what this is being used by (`tools/packaging/release/publish-kata-image.sh`?)

Comment (egernst) [reply tools/packaging/scripts/lib.sh:14]: You'll have to explain to me first :D

Comment (egernst) [reply tools/packaging/scripts/lib.sh:14]: Updated commit message -- please TAL and let me know if this is okay for you (ACK that I didn't make it a unique commit).

**Review by jodh-intel — APPROVED** [review]
That's @egernst.

**Review by likebreath — APPROVED** [review]
